cli/src/commands/interdiff.rs

```diff
@@ -27,25 +27,46 @@ use crate::complete;
 use crate::diff_util::DiffFormatArgs;
 use crate::ui::Ui;
 
-/// Compare the changes of two commits
+/// Show differences between the diffs of two revisions
 ///
-/// This excludes changes from other commits by temporarily rebasing `--from`
-/// onto `--to`'s parents. If you wish to compare the same change across
-/// versions, consider `jj evolog -p` instead.
+/// This is like running `jj diff -r` on each change, then comparing those
+/// results. It answers: "How do the modifications introduced by change A
+/// differ from the modifications introduced by change B?"
```

Comment:
nit: Say "revision A" and "revision B" instead since they don't have be different changes? For me it's almost always the same change (but I agree that your first example below is reasonable)